src/app/app.component.ts

```diff
@@ -7,4 +10,26 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'courses-app';
+  user = 'Admin';
+  /* courses:MockCourseDataType[] = []; */
+  courses:MockCourseDataType[] = mockedCoursesList;
```

Comment:
Let me to explain : ) The 2 types are not the same. The authors property has the names too in the CourseData while only the author ids in the Mocked type. Reason to distinguish is that we need the author names for the courses not just the ids in the children components -> We cannot pass simply the mockedCourses to the children components. ->     
2 options in my view: 
1. we can pass both the mocked courses and the mocked author data to the children components and do the mapping in CourseCard component where we need the names. -> It is a bit less efficient in my view as we do author id to -> name mapping more time ( for each card ), besides we need to pass also authors mocked data... ( which concept can result less data (there would be no author name data in coursedata) if there are many courses with same authors , but now we have only a few so current version seems more efficient in data size too)  
2. option, current: to make the mapping only once at the beginning in the app-component and pass only the data we need. In my view this second is more efficient both in data and function call... 
+ The picture I dont really understand as I pass not courses but coursesWithAuthorNames. Why is it courses there?
<app-courses [courses]="coursesWithAuthorNames"></app-courses>
![app-comp](https://github.com/user-attachments/assets/241f4c27-febd-4a3c-a867-0ac3f778bcaa)
+ I did not put this mapping into a service as it is needed now only once. 

Can I keep this way, or do you see this differnetly after my thoughts? Thank you!

Reply:
Well, I made the change on the print screen to demonstrate the type mismatch. That was the intention. `[courses]="coursesWithAuthorNames"` is yet again trick, even when the `courses` defined in the component .ts is different in type. I would definitely expect the same type declared, instantiated in component .ts as it will be used when handed over as an input to AppCoursesComponent. It would be more straightforward like this: `<app-courses [courses]="courses">` or sg. very similar, like `[courses]="courseList"`.
But to address the original problem: how about updating Author model and / or mocked author? I guess it is lot better to pass on some unused information (an author id for instance) even when it is not used, displayed on the screen. This way we will have consistent models on both CourseData and Author. I mean a mock should not define underlying data structure. Eventually, an author is an author, he will have an id, a name, etc. regardless of actual data or mocked.

Reply:
just one more explanation as I see it is just the names which are confusing , but I will change if the explanation is not satisfying enough : ):
So now the coursesWithAuthorNames (CourseData type) is just different from the courses (MockCourseData type) in the way that it holds the author names and author ids too! for each course. So CourseData is just a helper type I would like to use to send all the needed information to CoursesComponent. It is not less than Authors + Courses it is just already a prepared data which Courses and its children need. Benefits: 1. I need only one Input in CoursesComponent and its children. (dont need authors as the names are already packed) 2. i dont have to call the mappings which assigns the author ids with the author names on each CourseCardComponent but I just do this mapping only once in the root component at start (or i guess in later versions only once when it is changed but not for each CourseCard). If we would have several courses it could be much effective in my view.
-> so yes courses now is different but just in the way that coursesWithAuthorNames is an already prepared data holding both the author names and the author ids next to a course).
I dont see inconsistency because it does not alter anything. Maybe just the names are confusing as MockCourseData is actually the basic data structure which describes the data structure we get from the mock file or later from api requests.  
Sorry for the long story, just dont see the drawback of this helper type which can be used to pack only the necessary, optimal data with. Maybe there are and it is not a good practice, just want to understand why it can be wrong or unclear later.

Reply:
Sorry for the late response. I had issues with pushing changes to GitHub... So, I have pushed a new branch to demonstrate how you can achieve the same with less code, with a singe `Course` model.
Here: https://github.com/momanten/angular-fundamentals-template/tree/single_course_entity

I believe the confusions started of with mocking the authors. In a frontend application it is unlikely that you would receive only the author id's through some API. If this was the case, then you would need to keep track of the Author database for example in indexedDB. This is not the case for us, but you can be sure to get the the course list along with it's authors (entities) as well.
Please check how it is changes in mocking and how it makes any mapping irrelevant. 'date' or 'creationDate' refers to the very same information, just stick with one that suits you better. (Probably the project setup was confusing where both were used for no reason...)
Remember KISS principle, do more with less code!

Reply:
Thank you Tamas for the effort!
Now I see your point , I understand however I did not know we can cahnge the mock data structure. This way of course it is simpler however we will get the result from the Api requests in the other way. (where author name is not part of courses data) So it will look like as in my first solution. I need to do the mapping in the app component after the api requests. -> Should I do 1.this way (I will need to change again as in the router and API task we will get athors and courses as the initial mockdata defines taht )? 2.or should I do as it was initially , but using courses name as input parameters which is maybe not misleading.

Reply:
Probably we should not touch the mock data after all, although this should not define our development process. Since I am unsure what the intent behind this was, go for the 2nd version.